src/Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -1,47 +1,224 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.CommandLine;
 using System.CommandLine.Invocation;
 using System.IO;
+using System.Text;
 using System.Threading.Tasks;
 using Microsoft.DotNet.Interactive;
 using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.CSharp;
 using Microsoft.DotNet.Interactive.Events;
-using Microsoft.DotNet.Interactive.Formatting;
 
 namespace Microsoft.Data.Analysis.Interactive
 {
     public class DataFrameKernelExtension : IKernelExtension
     {
-        public Task OnLoadAsync(IKernel kernel)
+        bool _generateCsvMethod = false;
```

Comment:
Would it be better to flow this `bool` down through the method calls as a parameter, instead of setting a field property here? What happens if multiple threads are calling this object at the same time?